include/native/parallel_execution_native.h

```diff
@@ -32,105 +32,236 @@
 
 namespace grppi {
 
-//extern bool initialised;
-//extern thread_pool pool;
-/** @brief Set the execution mode to parallel with posix thread implementation 
- */
-struct parallel_execution_native {
-  public: 
-  thread_pool pool;
-  constexpr static int default_queue_size = 100;
-  int queue_size = default_queue_size;
-  queue_mode lockfree = queue_mode::blocking;
+/**
+\brief Thread index table to provide portable natural thread indices.
 
-  void set_queue_size(int new_size){
-     queue_size = new_size;
-  }
+A thread table provides a simple way to offer thread indices (starting from 0).
 
+When a thread registers itself in the registry. To get an integer index, users
```

Comment:
When a thread registers itself in the registry. I think that this sentence should be: "When a thread is created, it regirsters itself in the registry" or "When a thread is created, it is registered by itself in the registry.".